scheduler/src/main/java/org/cloudfoundry/autoscaler/scheduler/util/DateHelper.java

```diff
@@ -603,17 +602,9 @@
 	           "Pacific/Kiritimati"};
 
 	public static Date getDateWithZoneOffset(Date dateTime, TimeZone timeZone) {
-		SimpleDateFormat localSdf = new SimpleDateFormat(DATE_TIME_FORMAT);
-
-		SimpleDateFormat sdf = new SimpleDateFormat(DATE_TIME_FORMAT);
-		sdf.setTimeZone(timeZone);
-
-		try {
-			return sdf.parse(localSdf.format(dateTime));
-		} catch (ParseException e) {
-			throw new SchedulerInternalException("Invalid date specified. date=" + dateTime, e);
-		}
+		LocalDateTime local = LocalDateTime.ofInstant(dateTime.toInstant(), ZoneId.systemDefault());
```

Comment:
My point above was to simplify the code. We don't have to jump through so many hoops when ZonedDateTime would do it for us in fewer steps.